From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 14 Jul 2010 08:13:11 -0400
Subject: [wireless] rt2x00: use cancel_work_sync
Message-id: <20100714081306.4740.9700.send-patch@dhcp-lab-109.englab.brq.redhat.com>
Patchwork-id: 26878
O-Subject: [RHEL5 PATCH 4/7] rt2x00: use cancel_work_sync
Bugzilla: 582191
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>


diff --git a/drivers/net/wireless/rt2x00/rt2x00dev.c b/drivers/net/wireless/rt2x00/rt2x00dev.c
index f56081e..5b8b300 100644
--- a/drivers/net/wireless/rt2x00/rt2x00dev.c
+++ b/drivers/net/wireless/rt2x00/rt2x00dev.c
@@ -897,11 +897,7 @@ void rt2x00lib_remove_dev(struct rt2x00_dev *rt2x00dev)
 	/*
 	 * Stop all work.
 	 */
-#if 0 /* Not in RHEL5... */
 	cancel_work_sync(&rt2x00dev->intf_work);
-#else
-	ieee80211_cancel_work(rt2x00dev->hw, &rt2x00dev->intf_work);
-#endif
 
 	/*
 	 * Uninitialize device.
diff --git a/drivers/net/wireless/rt2x00/rt2x00link.c b/drivers/net/wireless/rt2x00/rt2x00link.c
index 910cd13..5167430 100644
--- a/drivers/net/wireless/rt2x00/rt2x00link.c
+++ b/drivers/net/wireless/rt2x00/rt2x00link.c
@@ -369,12 +369,7 @@ void rt2x00link_start_tuner(struct rt2x00_dev *rt2x00dev)
 
 void rt2x00link_stop_tuner(struct rt2x00_dev *rt2x00dev)
 {
-#if 0 /* Not in RHEL5... */
 	cancel_delayed_work_sync(&rt2x00dev->link.work);
-#else
-	if (delayed_work_pending(&rt2x00dev->link.work))
-		cancel_rearming_delayed_work(&rt2x00dev->link.work);
-#endif
 }
 
 void rt2x00link_reset_tuner(struct rt2x00_dev *rt2x00dev, bool antenna)
